driver/postgres/driver.go

```diff
@@ -0,0 +1,113 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/juju/errors"
+	_ "github.com/lib/pq" // import driver
+	"github.com/wallester/migrate/driver"
+	"github.com/wallester/migrate/file"
+)
+
+type postgres struct {
+	connection *sql.DB
+}
+
+// New returns new instance
+func New() driver.Driver {
+	return &postgres{}
+}
+
+// Open opens database connection
+func (db *postgres) Open(url string) error {
+	connection, err := sql.Open("postgres", url)
+	if err != nil {
+		return errors.Annotate(err, "connecting to database failed")
+	}
+
+	db.connection = connection
+
+	return nil
+}
+
+// Close closes database connection
+func (db *postgres) Close() error {
+	if err := db.connection.Close(); err != nil {
+		return errors.Annotate(err, "closing database connection failed")
+	}
+
+	return nil
+}
+
+// SelectAllMigrations selects existing migrations
+func (db *postgres) SelectAllMigrations(ctx context.Context) (map[int64]bool, error) {
+	rows, err := db.connection.QueryContext(ctx, `
+		SELECT version FROM schema_migrations
+	`)
+	if err != nil {
+		return nil, errors.Annotate(err, "selecting existing migration versions failed")
+	}
+
+	migrated := make(map[int64]bool)
+	for rows.Next() {
+		var version int64
+		if err := rows.Scan(&version); err != nil {
+			if closeErr := rows.Close(); closeErr != nil {
+				return nil, errors.Annotate(closeErr, "closing rows failed")
+			}
+
+			return nil, errors.Annotate(err, "scanning version failed")
+		}
+
+		migrated[version] = true
+	}
+
+	if closeErr := rows.Close(); closeErr != nil {
+		return nil, errors.Annotate(closeErr, "closing rows failed")
+	}
+
+	return migrated, nil
+}
+
+// CreateMigrationsTable creates migrations table if it does not exist yet
+func (db *postgres) CreateMigrationsTable(ctx context.Context) error {
+	if _, err := db.connection.ExecContext(ctx, `
+		CREATE TABLE IF NOT EXISTS schema_migrations(
+			version bigint not null primary key
+		)
+	`); err != nil {
+		return errors.Annotate(err, "creating schema_migrations table failed")
+	}
+
+	return nil
+}
+
+var applyMigrationSQL = map[bool]string{
+	true:  "INSERT INTO schema_migrations(version) VALUES($1)",
```

Comment:
can we have constants like up  = true and down = false, as such construction is not really readable?